File: Mathematical Algorithms/Dyck path/DyckPath.cpp
int countDyckPaths(unsigned int n) {
    // Compute value of 2nCn
    int res = 1;
    for (int i = 0; i < n; ++i) {
        res *= (2 * n - i);
        res /= (i + 1);
    }

    // return 2nCn/(n+1)
    return res / (n + 1);
}

File: Mathematical Algorithms/Dyck path/DyckPath.py
def countDyckPaths(n):
    # Compute value of 2nCn
    res = 1
    for i in range(0, n):
        res *= (2 * n - i)
        res /= (i + 1)

    # return 2nCn/(n+1)
    return res / (n + 1)